src/smooshable.cpp

```diff
@@ -75,6 +78,16 @@ SmooshableGermline::SmooshableGermline(
   assert(right_flex <= emission_indices.size());
   germline.MatchMatrix(start, emission_indices, left_flex, right_flex,
                        marginal_);
-  viterbi_ = marginal_;
+
+  // if all entries of 'marginal_' are less than SCALE_THRESHOLD, multiply
+  // all elements by SCALE_FACTOR. If not, do nothing.
+  if((marginal_.array() < SCALE_THRESHOLD).all()) {
```

Comment:
@dunleavy005 don't we want for this to be a while loop? That is, even after multiplying by SCALE_FACTOR everything could be below SCALE_THRESHOLD, motivating another round.

It also seems to be that extracting this out into a function would be good, because we'll have to do this operation in other settings (e.g. smooshing). It would take a matrix (an in/out argument) and then return an increment for the log scaler.